src/SeleniumLibrary/keywords/browsermanagement.py

```diff
@@ -119,6 +119,69 @@ def open_browser(self, url, browser='firefox', alias=None,
+        and for Firefox these
+        [https://seleniumhq.github.io/selenium/docs/api/py/webdriver_firefox/selenium.webdriver.firefox.options.html?highlight=firefox#selenium.webdriver.firefox.options.Options|methods and attributes]
+        are available. Please note that not all browsers supported by the
+        SeleniumLibrary have Selenium options available. Therefore please
+        consult the Selenium documentation which browsers do support
+        the Selenium options. Selenium options are also supported,
+        when ``remote_url`` argument is used.
+
+        The SeleniumLibrary ``options`` argument accepts Selenium
+        options in two different formats: as a string and as Python object
+        which is an instance of the Selenium options class.
+
+        The string format allows to define Selenium options methods
+        or attributes and their arguments in Robot Framework test data.
+        The method and attributes names are case and space sensitive and
+        must match to the Selenium options methods and attributes names.
+        When defining a method, is must defined in similar way as in
+        python: method name, opening parenthesis, zero to many arguments
+        and closing parenthesis. If there is need to define multiple
+        arguments for a single method, arguments must be separated with
+        comma, just like in Python. Example: `add_argument("--headless")`
+        or `add_experimental_option("key", "value")`. Attributes are
+        defined in similar way as in Python: attribute name, equal sing
+        and attribute value. Example, `headless=True`. Multiple methods
+        and attributes must separated by a semicolon, example:
+        `add_argument("--headless");add_argument("--start-maximized")`.
+
+        Arguments allow defining Python data types and arguments are
+        evaluated by using Python
+        [https://docs.python.org/3/library/ast.html#ast.literal_eval|ast.literal_eval].
+        Strings must be quoted with single or double quotes, example "value"
+        or 'value'. It is also possible define other Python builtin
+        data types, example `True` or `None`, by not using quotes
+        around the arguments.
+
+        The string format is space friendly and usually spaces do not alter
+        the defining the methods or attributes. There are two exceptions.
+        In some Robot Framework test data formats, two or more spaces are
+        considered as cell separator and instead of defining a single
+        argument, two or more arguments may be defined. Spaces in string
+        arguments are not removed and are left as is. Example
+        `add_argument ( "--headless" )` is same as
+        `add_argument("--headless")`. But `add_argument(" --headless ")` is
+        not same same as `add_argument ( "--headless" )`, because
+        spaces inside of quotes are not removed.
+
```

Comment:
One difficulty with supporting a string formatted options is being backwards compatible over time or change the rules of how the string is formatted. You might try out the formatting with several users to see how it works for them.

Reply:
That is a worry which I also have, but there are some safe guards. Calling the Selenium options is as much as possible backwards compatible, because behind the scenes it does a similar trick as the `Create WebDriver` keyword with Python `getattr`, `setattr` and `callable`, [more details in here](https://github.com/robotframework/SeleniumLibrary/pull/1393/files#diff-566132cdf5a534d7e99598e5acc77378R388). Therefore keyword leaves to the user to correctly defined and maintain the methods and attributes user wants to call.

But there is a part that worries me and that is the importing options. First Selenium options import path may change or the options class name may change. In the past, that part has been quite stable and I have not seen any changes in the upcoming Selenium 4.0 release either. But who knows what future will bring.

Also mobile devices, which are supported on the `Open Browser` are always done by using the  `webdriver.Remote`. The remote WebDriver supports options object, but those mobile devices don't have their own opinions object (Like example Chrome or Firefox has). Therefore the keyword does not support creating options object for those mobile devices by using the string format. Also there is feature/bug when Selenium options instance object is passed directly in, for mobile devices, [it will fail](https://github.com/robotframework/SeleniumLibrary/pull/1393/files#diff-566132cdf5a534d7e99598e5acc77378R353), but that feature can be lifted away. But there is some indication that for example Android might directly support Chrome options http://chromedriver.chromium.org/getting-started/getting-started---android Also I know that some users are using SeleniumLibrary for mobile testing because, according to some users feedback: "SeleniumLibrary has better keywords but is little bit tricky to setup." Therefore it might be interesting to support also options for mobile devices, but currently I lack the knowledge how that part actually works and the Travis testing infrastructure is not not setup (Although there seems to be [support for mobile](https://docs.travis-ci.com/user/languages/android/))

In any case, if the limitation in the string format is left as it is, it should be documented. Also we should remove the limitation when the Selenium options instance is passed in for mobile devices. That limitation should be easy to change, because now we support only two formats.

Reply:
It's impossible to make this fully future compatible because Selenium doesn't have a stable API for specifying options. The current approach to support specifying options like `method('arg')` and `attr = 'value'` has a benefit that users can adjust their code if Selenium changes.

Reply:
That is true. I still need to think should we do some extra for the mobile users. But perhaps easiest way is to: 
1) Fix the bug with mobile and Selenium options
2) Android == Chrome for selenium options. + Need documentation.